docs/transaction-verification.md

```diff
@@ -0,0 +1,169 @@
+ 
+#### Chain Update
+Before executing a chain update the following is always checked:
+ 
+1. `expiry` The transaction must never be expired and if it is (expired) then it will be deemed invalid.
+2. `sequential sequence number` It is verified that the sequence number of the transaction is the next one.
+If the `sequence number` is not the next one then the transaction is invalid and will not be executed.
+ 
+The `Scheduler` will now use the cached verification result computed in the `initial verification`. 
+If the transaction passed the `initial verification` but the signatures could not be validated then it is checked now and based on the result it will be deemed valid or invalid.
+ 
+On the other hand, if the transaction did pass the `initial verification` the cached verification result contains a hash of the public keys used for verifying the signature. It is asserted that this hash corresponds to the current set of authorized keys. If the authorized keys are the same then the transaction is deemed valid. Otherwise we re-verify the signature on the transaction as the authorized keys might have changed since the transaction was first processed, if the 
+signature can be verified the transaction is accepted otherwise it is rejected.
+ 
+#### Transaction
+Before executing a `normal transaction` the following is always checked:
+ 
+1. `remaining block energy` There must be enough energy left in the current block, if this is not the case then the transaction will be considered invalid.
+2. `expiry` The transaction may not be expired.
+3. `sender exists` The sender of the transaction must exist on the chain. If the sender does not exist then the transaction is deemed invalid.
+4. `senders balance` The sender must have enough funds to cover the transfer. If this is not the case then the transaction is deemed invalid.
+5. `sequential nonce` The _next_ account nonce of the sender account must match the nonce in the transaction. If not, then
+the transaction invalid.
+ 
+If the transaction was received as part of a block and did only partially pass the `initial verification` then the transaction is re-verified. This verification must pass _successfully_ otherwise the transaction is deemed invalid.
+If the transaction was _successfully_ passing the `initial verification`then it is checked that the keys at that point in time matches the current ones registered on chain. This is checked by using a cached hash of the keys used for verifying in the `initial verification` with the current keys for the account. If it turns out the keys match then the transaction is valid. Otherwise we re-verify the signature with the current registered keys. Nb. key(s) might've been added for the account in the meantime, but the old keys are still present and as such it makes sense to re-verify the signature given the current account keys and thus the transaction is either deemed valid or invalid.
```

Comment:
It is also important that the threshold is considered as part of the keys. I'd recommend adding.

Reply:
Rephrasing this part together with `ChainUpdate`. Now using more correct terms as `AccountInformation` and `UpdateKeysCollection`.